fw/http_parser.c

```diff
@@ -902,15 +923,15 @@ process_trailer_hdr(TfwHttpMsg *hm, TfwStr *hdr, unsigned int id)
 		if (chunk->len == (str)->len) {				\
 			lambda;						\
 			TRY_STR_INIT();					\
-			__FSM_I_MOVE_fixup_f(next_st, __fsm_n, field, 0);\
+			__FSM_I_MOVE_fixup_f(matched_st, __fsm_n, field, 0);\
```

Comment:
Agree about my bad wording, "on_match_st" sounds better. Those are internal identifiers so they mostly function as documentation, and I found this documentation non-descriptive. "Next state" means "the one that goes after the current one" - but that is the same `curr_st` most of the time, because we put several `TRY_STR_*()` one after another.

Also, things get worse at `H2_TRY_STR_*` family of functions e.g. `H2_TRY_STR_LAMBDA_BY_REF_finish(str, lambda1, lambda2, finish, state)` - those numbered arguments tell absolutely nothing about how to use them. I haven't tried to elaborate on the latter problem because I don't really understand why `lambda1` is different from `lambda2`.

Reply:
As I commented above - don't introduce renaming if you don't change the logic in this patch.

In general it's a good practice to separate renaming and refactoring from bug fixes and features